src/Elliptic/Systems/IrrotationalBns/Equations.cpp

```diff
@@ -0,0 +1,149 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/IrrotationalBns/Equations.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/PrefixHelpers.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Identity.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/Systems/IrrotationalBns/Geometry.hpp"
+#include "Elliptic/Systems/Poisson/Equations.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace IrrotationalBns {
+
+void flat_potential_fluxes(
+    const gsl::not_null<tnsr::I<DataVector, 3>*> flux_for_potential,
+    const tnsr::II<DataVector, 3>& rotational_shift_stress,
+    const tnsr::i<DataVector, 3>& auxiliary_velocity) {
+  ::tenex::evaluate<ti::I>(
+      flux_for_potential,
+      -auxiliary_velocity(ti::j) * rotational_shift_stress(ti::I, ti::J));
+  for (size_t d = 0; d < 3; d++) {
+    flux_for_potential->get(d) += auxiliary_velocity.get(d);
+  }
+}
+
+void curved_potential_fluxes(
+    const gsl::not_null<tnsr::I<DataVector, 3>*> flux_for_potential,
+    const tnsr::II<DataVector, 3>& rotational_shift_stress,
+    const tnsr::II<DataVector, 3>& inverse_spatial_metric,
+    const tnsr::i<DataVector, 3>& auxiliary_velocity) {
+  ::tenex::evaluate<ti::I>(flux_for_potential,
+                           (inverse_spatial_metric(ti::I, ti::J) -
+                            rotational_shift_stress(ti::I, ti::J)) *
+                               auxiliary_velocity(ti::j));
+}
+
+void fluxes_on_face(
+    gsl::not_null<tnsr::I<DataVector, 3>*> face_flux_for_potential,
+    const tnsr::i<DataVector, 3>& face_normal,
+    const tnsr::I<DataVector, 3>& face_normal_vector,
+    const tnsr::II<DataVector, 3>& rotational_shift_stress,
+    const Scalar<DataVector>& velocity_potential) {
+  ::tenex::evaluate<ti::I>(
+      face_flux_for_potential,
+      velocity_potential() *
+          (face_normal_vector(ti::I) -
+           rotational_shift_stress(ti::I, ti::J) * face_normal(ti::j)));
```

Comment:
This quantity `B^{ij} n_j` _could_ be precomputed as an optimization. Maybe just add a comment here for the possible performance optimization.